File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        
        int left = 0;
        int right = x;
        int mid;
        long square;
        
        while (left <= right) {

            mid = (left + right) / 2;
            square = (long) mid * mid;
            if (x == square) {
                return mid;
            } else if (x < square) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
            
        }
        return left - 1;
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:

        if x == 0:
            return 0
        
        left = 0
        right = x+1

        while left < right:
            mid = (left + right) // 2
            if (mid * mid > x):
                right = mid
            else:
                left = mid + 1
        
        return left - 1